analyzers/src/SonarAnalyzer.VisualBasic/Rules/TooManyLabelsInSwitch.cs

```diff
@@ -18,31 +18,39 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-namespace SonarAnalyzer.Rules.VisualBasic
+namespace SonarAnalyzer.Rules.VisualBasic;
+
+[DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+public sealed class TooManyLabelsInSwitch : TooManyLabelsInSwitchBase<SyntaxKind, SelectStatementSyntax>
 {
-    [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
-    public sealed class TooManyLabelsInSwitch : TooManyLabelsInSwitchBase<SyntaxKind, SelectStatementSyntax>
-    {
-        protected override DiagnosticDescriptor Rule { get; } =
-            DescriptorFactory.Create(DiagnosticId, MessageFormat,
-                isEnabledByDefault: false);
+    private static readonly ISet<SyntaxKind> IgnoredStatementsInSwitch = new HashSet<SyntaxKind> { SyntaxKind.ReturnStatement, SyntaxKind.ThrowStatement };
+
+    protected override DiagnosticDescriptor Rule { get; } =
+        DescriptorFactory.Create(DiagnosticId, string.Format(MessageFormat, "Select Case", "Case"),
+            isEnabledByDefault: false);
 
-        private const string MessageFormat = "Consider reworking this 'Select Case' to reduce the number of 'Case's" +
-            " from {1} to at most {0}.";
+    protected override SyntaxKind[] SyntaxKinds { get; } = [SyntaxKind.SelectStatement];
 
-        protected override SyntaxKind[] SyntaxKinds { get; } =
-            new[] { SyntaxKind.SelectStatement };
+    protected override GeneratedCodeRecognizer GeneratedCodeRecognizer =>
+        VisualBasicGeneratedCodeRecognizer.Instance;
 
-        protected override GeneratedCodeRecognizer GeneratedCodeRecognizer =>
-            VisualBasicGeneratedCodeRecognizer.Instance;
+    protected override SyntaxNode GetExpression(SelectStatementSyntax statement) =>
+        statement.Expression;
 
-        protected override SyntaxNode GetExpression(SelectStatementSyntax statement) =>
-            statement.Expression;
+    protected override int GetSectionsCount(SelectStatementSyntax statement) =>
+        ((SelectBlockSyntax)statement.Parent).CaseBlocks.Count;
 
-        protected override int GetSectionsCount(SelectStatementSyntax statement) =>
-            ((SelectBlockSyntax)statement.Parent).CaseBlocks.Count;
+    protected override bool AllSectionsAreOneLiners(SelectStatementSyntax statement) =>
+        ((SelectBlockSyntax)statement.Parent).CaseBlocks.All(HasOneLine);
+
+    protected override Location GetKeywordLocation(SelectStatementSyntax statement) =>
+        statement.SelectKeyword.GetLocation();
+
+    private static bool HasOneLine(CaseBlockSyntax switchSection)
+    {
+        var statements = switchSection.Statements.Where(x => !x.IsAnyKind(IgnoredStatementsInSwitch)).ToArray();
```

Comment:
Are BlockSyntax and IfStatement good enough? 
It could be anything: a switch, a loop, etc...

I doubt maintaining a statement of transparent syntax is a good idea, I would rather not.

Reply:
Doing so has the advantage of removing an FN.
I will add it, but I am not sure of the transparent syntax list.